src/main/java/com/google/sps/data/ProcessCareerQuizResults.java

```diff
@@ -0,0 +1,43 @@
+package com.google.sps.data;
+
+import java.util.List;
+import java.util.HashMap; 
+import java.util.Map; 
+import java.util.Map.Entry;
+import com.google.sps.data.CareerQuestionChoice;
+
+/** Returns the recommended career path based off the career quiz results. */
+public class ProcessCareerQuizResults {
+  private ProcessCareerQuizResults() {};
+  
+  public static String getRecommendedCareerPath(List<CareerQuestionChoice> userChoices) {
+    Map<String, Integer> choiceFrequencyHashMap = new HashMap<String, Integer>(); 
+    String careerPath;
+    String maxFrequencyCareerPath = new String();
+    // creates a frequency hashmap mapping associated career paths of choices to their frequencies
+    for (CareerQuestionChoice choice : userChoices) {
+      careerPath = choice.getAssociatedCareerPath();
+      if (choiceFrequencyHashMap.containsKey(careerPath)) {
+        choiceFrequencyHashMap.put(careerPath, choiceFrequencyHashMap.get(careerPath) + 1); 
+      } 
+      else { 
+        choiceFrequencyHashMap.put(careerPath, 1);
+      } 
+    }
+    return mostFrequentInHashMap(choiceFrequencyHashMap);
+  }
+
+  private static String mostFrequentInHashMap(Map<String, Integer> hashMap) {
+    int maxFrequencyCount = 0;
+    int currentFrequency;
+    String mostFrequentKey = "";
+    for (String key : hashMap.keySet()) {
+      currentFrequency = hashMap.get(key);
+      if (currentFrequency > maxFrequencyCount) {
+        mostFrequentKey = key;
+        maxFrequencyCount = currentFrequency;
+      }
+    }
+    return mostFrequentKey;
+  }
```

Comment:
This can be reduced to something like so:
```
    Entry<K, V> maxEntry = Collections.max(map.entrySet(), (Entry<K, V> e1, Entry<K, V> e2) -> e1.getValue()
        .compareTo(e2.getValue()));
    return maxEntry.getValue();
```

You can choose to do this, or keep what you have. It's more about clarity at this point.

Reply:
Very cool! I might keep what I have just for clarity.